js/private/js_image_layer.bzl

```diff
@@ -500,18 +531,45 @@ js_image_layer_lib = struct(
             default = "0:0",
         ),
         "compression": attr.string(
-            doc = "Compression algorithm. Can be one of `gzip`, `none`.",
-            values = ["gzip", "none"],
+            doc = "Compression algorithm. See https://github.com/bazel-contrib/bazel-lib/blob/bdc6ade0ba1ebe88d822bcdf4d4aaa2ce7e2cd37/lib/private/tar.bzl#L29-L39",
+            values = tar_lib.common.accepted_compression_types + ["none"],
             default = "gzip",
         ),
         "platform": attr.label(
             doc = "Platform to transition.",
         ),
         "generate_empty_layers": attr.bool(
-            doc = """Generate layers even if they are empty.
+            # TODO(3.0): remove this attribute.
+            doc = """DEPRECATED. An empty layer is always generated if the layer group have no matching files.""",
+            default = False,
+        ),
+        "layer_groups": attr.string_dict(
+            doc = """Layer groups to create.
+These are utilized to categorize files into distinct layers, determined by their respective paths. 
+The expected format for each entry is "<key>": "<value>", where <key> MUST be a valid Bazel and 
+JavaScript identifier (alphanumeric characters),  and <value> MAY be either an empty string (signifying a universal match)
+or a valid regular expression.
+ 
+For better performance, it is recommended to split the large parts of a js_binary to have a separate layer.
```

Comment:
What does "large parts of a js_binary" mean? Does that mean the `js_binary(data)` such as transitive js+npm paths?

Reply:
Anything, its up to the user to determine if something "large" and should go into its own layer.